test/integration/client-side-encryption/client_side_encryption.prose.14.decrytion_events.test.ts

```diff
@@ -0,0 +1,252 @@
+    const buffer = Buffer.from(cipherText.buffer);
+    buffer.writeInt8(0, buffer.length - 1);
+    malformedCiphertext = new Binary(buffer, 6);
+    // Create a MongoClient named ``encryptedClient`` with these ``AutoEncryptionOpts``:
+    //   AutoEncryptionOpts {
+    //     keyVaultNamespace: "keyvault.datakeys";
+    //     kmsProviders: { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } }
+    //   }
+    // Configure ``encryptedClient`` with "retryReads=false".
+    encryptedClient = this.configuration.newClient(
+      {},
+      {
+        writeConcern: { w: 'majority' },
+        retryReads: false,
+        monitorCommands: true,
+        autoEncryption: {
+          keyVaultNamespace: 'keyvault.datakeys',
+          kmsProviders: { local: { key: LOCAL_KEY } },
+          extraOptions: getEncryptExtraOptions()
+        }
+      }
+    );
+    // Register a listener for CommandSucceeded events on ``encryptedClient``.
+    encryptedClient.on('commandSucceeded', event => {
+      if (event.commandName === 'aggregate') {
+        aggregateSucceeded = event;
+      }
+    });
+    // The listener must store the most recent CommandFailedEvent error for the "aggregate" command.
+    encryptedClient.on('commandFailed', event => {
+      if (event.commandName === 'aggregate') {
+        aggregateFailed = event;
+      }
+    });
+  });
+
+  afterEach(async function () {
+    aggregateSucceeded = undefined;
+    aggregateFailed = undefined;
+    await setupClient.close();
+    await encryptedClient.close();
+  });
+
+  context('Case 1: Command Error', metadata, function () {
+    beforeEach(async function () {
+      // Use ``setupClient`` to configure the following failpoint:
+      //    {
+      //         "configureFailPoint": "failCommand",
+      //         "mode": {
+      //             "times": 1
+      //         },
+      //         "data": {
+      //             "errorCode": 123,
+      //             "failCommands": [
+      //                 "aggregate"
+      //             ]
+      //         }
+      //     }
+      await setupClient
+        .db()
+        .admin()
+        .command({
+          configureFailPoint: 'failCommand',
+          mode: {
+            times: 1
+          },
+          data: {
+            errorCode: 123,
+            failCommands: ['aggregate']
+          }
+        });
+    });
+
+    it('expects an error and a command failed event', async function () {
+      // Use ``encryptedClient`` to run an aggregate on ``db.decryption_events``.
+      // Expect an exception to be thrown from the command error. Expect a CommandFailedEvent.
+      const collection = encryptedClient.db('db').collection('decryption_events');
+
+      const error = await collection
+        .aggregate([])
+        .toArray()
+        .catch(error => error);
+
+      expect(error).property('code', 123);
+      expect(aggregateFailed).nested.property('failure.code', 123);
```

Comment:
```suggestion
      expect(error).to.have.property('code', 123);
      expect(aggregateFailed).to.have.nested.property('failure.code', 123);
```

minor but we generally try for natural phrasing in our assertions

Reply:
(there are a number of assertions like this in this file that are new)